rs/utils/validate_eq_derive/src/lib.rs

```diff
@@ -0,0 +1,150 @@
+extern crate proc_macro;
+extern crate quote;
+extern crate syn;
+
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::Data::Struct;
+
+/// Derive of ValidateEq trait.
+/// Derived implementation compares fields of lhs and rhs and returns the first divergence if any.
+/// A field can have one #[validate_eq(Recursive|Skip)] attribute.
+///   - Recursive calls .validate_eq() and returns path + its error in case of divergence
+///   - Skip ignores the field.
+///   - None (default) compares fields using PartialEq and reports their name in case of
+///     divergence.
+
+enum ValidateEqFieldAttr {
+    /// Compare using .eq() and return field name if diverges.
+    PartialEq,
+    /// Call .validate_eq(); in case of deivergence return the field name and the underlying
+    /// divergence error string.
+    Recursive,
+    /// Ignore for ValidateEq
+    Skip,
+}
+
+// Find #[validate_eq(...)] attribute if any.
```

Comment:
Nit: There's no good reason for this not to be a doc comment.
```suggestion
/// Find #[validate_eq(...)] attribute if any.
```